src/client/sandbox/child-window/index.ts

```diff
@@ -88,10 +88,44 @@ export default class ChildWindowSandbox extends SandboxBase {
 
             e.preventDefault();
 
-            this._openUrlInNewWindow(url);
+            this._openNewWindowAfterAllEventHandlers(url, e);
         });
     }
 
+    private _openNewWindowAfterAllEventHandlers (url, e) {
+        const nativeAddEventListener    = nativeMethods.windowAddEventListener || nativeMethods.addEventListener;
+        const nativeRemoveEventListener = nativeMethods.windowRemoveEventListener || nativeMethods.removeEventListener;
+
+        let eventBubbledToTop  = false;
+        let isDefaultPrevented = false;
+
+        const openUrlInNewWindowIfNotPrevented = () => {
+            eventBubbledToTop = true;
+
+            nativeRemoveEventListener.call(window, 'click', openUrlInNewWindowIfNotPrevented);
+
+            if (!isDefaultPrevented)
+                this._openUrlInNewWindow(url);
+        };
+
+        nativeAddEventListener.call(window, 'click', openUrlInNewWindowIfNotPrevented);
+
+        // NOTE: additional attempt to open a new window if window.handler was prevented by
+        // `stopPropagation` or `stopImmediatePropagation` methods
+        const origPreventDefault = e.preventDefault;
+
+        e.preventDefault = () => {
+            isDefaultPrevented = true;
+
+            return origPreventDefault.call(e);
+        };
+
+        setTimeout(() => {
```

Comment:
Add the `delay(fn, timeout)` utility function.

Reply:
We already have the `nextTick` method here:
https://github.com/DevExpress/testcafe-hammerhead/blob/master/src/client/utils/next-tick.ts